<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="View" name="graphicsView"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionCreate"/>
   <addaction name="actionTransform"/>
   <addaction name="separator"/>
   <addaction name="actionSelect"/>
   <addaction name="actionRotate"/>
   <addaction name="actionTranslate"/>
   <addaction name="actionScale"/>
  </widget>
  <action name="actionTransform">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Transform</string>
   </property>
   <property name="toolTip">
    <string>Transform</string>
   </property>
   <property name="shortcut">
    <string>2</string>
   </property>
  </action>
  <action name="actionRotate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Rotate</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionTranslate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Translate</string>
   </property>
   <property name="shortcut">
    <string>T</string>
   </property>
  </action>
  <action name="actionScale">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Scale</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="actionCreate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Create</string>
   </property>
   <property name="shortcut">
    <string>1</string>
   </property>
  </action>
  <action name="actionSelect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Select</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>View</class>
   <extends>QGraphicsView</extends>
   <header>View.hpp</header>
   <slots>
    <slot>setTransformEditMode()</slot>
    <slot>setRotateTransformMode()</slot>
    <slot>setTranslateTransformMode()</slot>
    <slot>setScaleTransformMode()</slot>
    <slot>setCreateEditMode()</slot>
    <slot>setSelectTransformMode()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionCreate</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setCreateMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRotate</sender>
   <signal>triggered()</signal>
   <receiver>graphicsView</receiver>
   <slot>setRotateTransformMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionScale</sender>
   <signal>triggered()</signal>
   <receiver>graphicsView</receiver>
   <slot>setScaleTransformMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTransform</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>setTransformMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTranslate</sender>
   <signal>triggered()</signal>
   <receiver>graphicsView</receiver>
   <slot>setTranslateTransformMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSelect</sender>
   <signal>triggered()</signal>
   <receiver>graphicsView</receiver>
   <slot>setSelectTransformMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setCreateMode()</slot>
  <slot>setTransformMode()</slot>
 </slots>
</ui>
